app/mailers/step_mailer.rb

```diff
@@ -3,6 +3,15 @@ def step_reply_received(step)
     @proposal = step.proposal.decorate
     assign_threading_headers(@proposal)
     @step = step
+    last_approved_user = nil
+    
+    # Iterate through steps to find the LAST 'approved' person
+    @proposal.individual_steps.each_with_index do |step, index|
+      if step.status == 'approved'
+        last_approved_user = User.find step.user_id
+      end
+    end
```

Comment:
additional thought: could include this logic in `app/models/concerns/step_manager.rb` (which we use in `Proposal`) so we could have a unit test for this